agent/agent_test.go

```diff
@@ -626,6 +628,71 @@ func TestAgent_RemoveServiceRemovesAllChecks(t *testing.T) {
 	}
 }
 
+func TestAgent_AddCheck_RemoveServiceRace(t *testing.T) {
```

Comment:
IMO, I'd drop the test since the entire data structure of the `checkXXX` maps with the partial state and the lock handling is broken. It is great that you found it but the correct fix will come with #3294 and we don't want to drag this test along.